### PR TITLE
Merged NuttX two debugging tutorials

### PR DESCRIPTION
**Pull request by tkolcon**
I merged two tutorials created by Ingo according to: "Debugging a NuttX target - 
Merge the two existing tutorials". Please check if I modified all necessary files.

### CODE REVIEW

File: _docs/tutorials/advanced/nuttx/debugging/index.md [hunk 1]
```diff
@@ -1,33 +1,37 @@
 ---
-title: Debugging a NuttX target with GDB and OpenOCD
-permalink: /docs/tutorials/advanced/nuttx/debugging_gdb_openocd/
-author: Ingo Lütkebohle
+title: Debugging a NuttX
+permalink: /docs/tutorials/advanced/nuttx/debugging/
+author: Ingo Lütkebohle (merged by Tomasz Kołcon)
 ---
 
+This tutorial consists of 2 parts: debugging a NuttX target with GDB and OpenOCD and debugging with Visual Studio Code. The first one describes tools installation and debugging via command line. The second focuses on the description of debugging using the modern IDE.
```
Comment (ralph-lange) [inline _docs/tutorials/advanced/nuttx/debugging/index.md:7]:
```suggestion
This tutorial consists of two parts: First, debugging a NuttX target with GDB and OpenOCD. This part covers tool installation and debugging via command line. Second, debugging with Visual Studio Code, i.e. using a modern IDE.
```

File: _docs/tutorials/advanced/nuttx/debugging/index.md [hunk 2]
```diff
@@ -1,33 +1,37 @@
 ---
-title: Debugging a NuttX target with GDB and OpenOCD
-permalink: /docs/tutorials/advanced/nuttx/debugging_gdb_openocd/
-author: Ingo Lütkebohle
+title: Debugging a NuttX
```
Comment (ralph-lange) [inline _docs/tutorials/advanced/nuttx/debugging/index.md:2]:
```suggestion
title: Debugging a NuttX Application
```

File: _docs/tutorials/advanced/nuttx/debugging/index.md [hunk 3]
```diff
@@ -128,7 +132,7 @@ This switches to thread 2 and then inspects the local variables, of which there
 In my case, this is the NSH thread which is waiting for some input.
 
 
-## Conclusion
+### Conclusion
```
Comment (ralph-lange) [inline _docs/tutorials/advanced/nuttx/debugging/index.md:135]: I propose to remove the conclusion subsubsection here completely.

File: _docs/tutorials/advanced/nuttx/debugging/index.md (hunk 2 above)
Comment (tkolcon) [reply _docs/tutorials/advanced/nuttx/debugging/index.md]: Done.

**Pull request merged**